### PR TITLE
Introduce strings template param to force add file name enum

### PR DESCRIPTION
**Pull request by viktorasl**
* [x] I've started my branch from the `develop` branch (gitflow)
  * [x] And I've selected `develop` as the "base" branch for this Pull Request I'm about to create
* [x] I've added an entry in the `CHANGELOG.md` file to explain my changes and credit myself  
      _(or added `#trivial` to the PR title if I think it doesn't warrant a mention in the CHANGELOG)_
  * [x] Add the entry in the appropriate section (Bug Fix / New Feature / Breaking Change / Internal Change)
  * [x] Add a period and 2 spaces at the end of your short entry description
  * [x] Add links to your GitHub profile to credit yourself and to the related PR and issue after your description

---

During migration my team and myself moved some files which used `L10n` structure to other frameworks, after moving `.strings` file there as well we are not be able to use the same `L10n.<FileName>` structure as there is only a single `.strings` file there.

As a solution I've introduced a new file template, but the change is simple, so to keep consistency in those kind of situations I'd like to suggest an option to force add file name `enum`s.

* [x] Documentation updated
* [x] Not a breaking change, option not being provided defaults to old behaviour
* [x] Updated tests
* [x] Not added support for swift3

Cheers! ☮️

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -51,6 +51,9 @@
   [David Jennes](https://github.com/djbe)
   [#595](https://github.com/SwiftGen/SwiftGen/issue/595)
   [#600](https://github.com/SwiftGen/SwiftGen/pull/600)
+* Strings: template option to force add file name `enum` structure even if only single file is provided (unsupported for swift3).
```
Comment (SwiftGen-Eve) [inline CHANGELOG.md]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/SwiftGen/SwiftGen/blob/82ca1c71fe58e4bc7acb0a1e9d25d2077fbaf41e/CHANGELOG.md#L54"></span>Line describing your entry should end with a period and 2 spaces.</td>
    </tr>
  </tbody>
</table>

**Comment by SwiftGen-Eve** [comment]
Hey 👋 I'm Eve, the friendly bot watching over SwiftGen 🤖

Thanks a lot for your contribution!



---

Seems like everything is in order 👍 You did a good job here! 🤝

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Tests/TemplatesTests/StringsTests.swift [hunk 2]
```diff
@@ -29,10 +29,19 @@ class StringsTests: XCTestCase {
...
     ]
   }
 
+  private lazy var variationsSwift3: VariationGenerator = { name, context in
+    let notSupported = ["-forceFileNameEnum"]
+    return try self.variations(name, context).filter {
+      !notSupported.contains($0.suffix)
+    }
+  }
+
```
Comment (AliSoftware) [inline Tests/TemplatesTests/StringsTests.swift]: We actually thought about dropping/removing the Swift 3 templates for the next release… If we decide to do so, such a special case in tests would not be needed anymore – \cc @djbe

**Review by AliSoftware — APPROVED** [review]

**Comment by AliSoftware** [comment]
Before merging this PR we'll need to:
 - [x] Merge #691 (obsolete Swift 3 templates)
 - [x] Rebase this PR on top of `develop` (this should also fix the CI failure currently present in this PR due to a change in our CI config recently merged)
 - [x] Update this PR to remove the special cases that have been introduced for Swift 3 (rightfully at that time, but they won't be necessary once we drop Swift 3 after #691)

**Comment by AliSoftware** [comment]
Thanks again @viktorasl for the PR! Merging now.

Our next step is to create another PR to apply that same change to other templates (an not just for `strings` template) 👍

**Pull request merged**